### PR TITLE
docs

### PR DESCRIPTION
**Pull request by 0xJepsen**
This PR gives some tlc to our docs by editing our readme, moving a good chunk of it into the kzg readme, and adding a tiny RSA readme.

### CODE REVIEW

File: src/curve/README.md [hunk 1]
```diff
@@ -58,6 +58,43 @@ $f_{r,P}$ can be calculated as $f_{r-1,P}\cdot\frac{l_{[r-1]P,P}}{v_{rP}}$ where
...
+Curve endomorphisms are maps that take points in one a curve subgroup and map them to themselves. An example is point addition and point doubling. The Frobinious endomorphism denoted $\Phi$  takes points in $P \in E(F_q)$ and maps them $\Phi(P) = (X^q, Y^q)$ .
+
+For higher powers of  the map you write $\Phi^k$ .
+
+### Trace Map
+
+This then allows us to define the trace map which takes points in $E(F_{q^k})$ and maps them to $E(F_q)$
+
+$$
+tr(P) = P + Φ(P ) + ... + Φ^{k−1}(P )
```
[the remark this answers is not in the document]
Comment (Autoparallel) [reply src/curve/README.md:85]: also use the '\Phi` for math, don't use the UTF8

**Review by Autoparallel — APPROVED** [review]
This is looking really good!! Most of my feedback is nit level (but I do think going through this with a fine tooth comb would make it shine). Adding more to RSA would be good. Also I think you could just call it `rsa` and not `tiny_rsa` as I'm not sure tiny adds anything

File: README.md [hunk 2]
```diff
@@ -16,72 +16,29 @@
...
-
-
+We have found the following resources helpful in understanding the foundational mathematics behind this implementation. After going through these, you should be able to understand the codebase
 
 ### Theoretic Resources
 - [A gentle introduction to Fast Fourier Transforms over Finite Fields](https://vitalik.eth.limo/general/2019/05/12/fft.html)
+- [Ben Lynn's Docatoral Thesis on the Implementation of Pairing-Based Cryptosystems](https://crypto.stanford.edu/pbc/thesis.pdf)
 - [Introduction to Pairings](https://vitalik.eth.limo/general/2017/01/14/exploring_ecp.html)
 - [KZG introduction by dankrad](https://dankradfeist.de/ethereum/2020/06/16/kate-polynomial-commitments.html)
 - [Pairings in depth](https://static1.squarespace.com/static/5fdbb09f31d71c1227082339/t/5ff394720493bd28278889c6/1609798774687/PairingsForBeginners.pdf)
```
[the remark this answers is not in the document]
Comment (0xJepsen) [reply README.md:44]: wait wasn't this your favorite resource?

File: src/tiny_rsa/README.md
[the remark this answers is not in the document]
Comment (0xJepsen) [reply src/tiny_rsa/README.md:1]: I will add in later PR

**Pull request merged**